### PR TITLE
Verify OE remote attestation on pre-production platforms

### PR DESCRIPTION
**Pull request by yanxue-intel**
Signed-off-by: yan xue <yan.xue@intel.com>

### CODE REVIEW

File: docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md [hunk 1]
```diff
@@ -275,3 +275,37 @@ Total Test time (real) =  83.61 sec
...
+-    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi71OiO\n"
+-    "SLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlA==\n"
++    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET/oP/VYc2tbA+Y0wjIEoxbknonMy\n"
++    "yOjrE/a+QrVx1kZvU8ZE/8L/wQKCIOSaSWbPAvMuL7TTSbssuu0okDegLQ==\n"
+     "-----END PUBLIC KEY-----\n";
+
+host/sgx/sgxquoteprovider.c
+ // Define the name of CA
+-static uint8_t CRL_CA_PROCESSOR[] = "processor";
++static uint8_t CRL_CA_PROCESSOR[] = "platform";
```
Comment (CodeMonkeyLeet) [inline docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md:311]: The formatting for this is unreadable in markdown format. Please enclose this block with the `` ```patch`` markdown so that it renders correctly. For example:

````
```patch
common/sgx/quote.c
 // Public key of Intel's root certificate.
 static const char* g_expected_root_certificate_key =
     "-----BEGIN PUBLIC KEY-----\n"
-    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi71OiO\n"
-    "SLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlA==\n"
+    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET/oP/VYc2tbA+Y0wjIEoxbknonMy\n"
+    "yOjrE/a+QrVx1kZvU8ZE/8L/wQKCIOSaSWbPAvMuL7TTSbssuu0okDegLQ==\n"
     "-----END PUBLIC KEY-----\n";

common/sgx/tcbinfo.c
 // Public key of Intel's root certificate.
 static const char* _trusted_root_key_pem =
     "-----BEGIN PUBLIC KEY-----\n"
-    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi71OiO\n"
-    "SLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlA==\n"
+    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET/oP/VYc2tbA+Y0wjIEoxbknonMy\n"
+    "yOjrE/a+QrVx1kZvU8ZE/8L/wQKCIOSaSWbPAvMuL7TTSbssuu0okDegLQ==\n"
     "-----END PUBLIC KEY-----\n";

host/sgx/sgxquoteprovider.c
 // Define the name of CA
-static uint8_t CRL_CA_PROCESSOR[] = "processor";
+static uint8_t CRL_CA_PROCESSOR[] = "platform";
```
````

File: docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md [hunk 2]
```diff
@@ -275,3 +275,37 @@ Total Test time (real) =  83.61 sec
...
+ // Public key of Intel's root certificate.
+ static const char* _trusted_root_key_pem =
+     "-----BEGIN PUBLIC KEY-----\n"
+-    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi71OiO\n"
+-    "SLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlA==\n"
++    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET/oP/VYc2tbA+Y0wjIEoxbknonMy\n"
++    "yOjrE/a+QrVx1kZvU8ZE/8L/wQKCIOSaSWbPAvMuL7TTSbssuu0okDegLQ==\n"
+     "-----END PUBLIC KEY-----\n";
+
+host/sgx/sgxquoteprovider.c
```
Comment (yentsanglee) [inline docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md:308]: Line 309-311 is not needed. PR 3294 fixed it.

File: docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md [hunk 3]
```diff
@@ -275,3 +275,37 @@ Total Test time (real) =  83.61 sec
 ```
 
 A clean pass of the above unit tests is an indication that your Open Enclave setup was successful.
+
+## 6. (Optional) Verify OE remote attestation on pre-production platforms
```
Comment (yentsanglee) [inline docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md:279]: Please state why and when the optional steps are needed.

File: docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md [hunk 4]
```diff
@@ -275,3 +275,37 @@ Total Test time (real) =  83.61 sec
...
+ // Public key of Intel's root certificate.
+ static const char* g_expected_root_certificate_key =
+     "-----BEGIN PUBLIC KEY-----\n"
+-    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEC6nEwMDIYZOj/iPWsCzaEKi71OiO\n"
+-    "SLRFhWGjbnBVJfVnkY4u3IjkDYYL0MxO4mqsyYjlBalTVYxFP2sJBK5zlA==\n"
++    "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAET/oP/VYc2tbA+Y0wjIEoxbknonMy\n"
++    "yOjrE/a+QrVx1kZvU8ZE/8L/wQKCIOSaSWbPAvMuL7TTSbssuu0okDegLQ==\n"
+     "-----END PUBLIC KEY-----\n";
+
+common/sgx/tcbinfo.c
```
Comment (jvega10) [inline docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md:298]: is this change needed to verify the signature of the tcb?
Example:
```
{
        "fmspc": "00606a000000",
        "tcbinfo":{
          "version": 2,
          "issueDate": "2020-05-07T15:07:45Z",
          "nextUpdate": "2020-06-06T15:07:45Z",
          "fmspc": "00606a000000",
          "pceId": "0000",
          "tcbType": 0,
          "tcbEvaluationDataNumber": 5,
          "tcbLevels":[
            {
              "tcb":{
               ...
              },
              "tcbDate":"2020-03-22T00:00:00Z",
              "tcbStatus":"UpToDate"
            },
            {
              "tcb":{
                ...
              },
              "tcbDate":"2020-03-22T00:00:00Z",
              "tcbStatus":"OutOfDate"
            }
          ]
        },
        "signature":"d6a66aa147c87d38756bb97dc5c25808f8685d99fadb006d297bbfed5b2f253948240eae343915ce325c3334fe71d28595ef0a98798a0491ea2df3dfda21f0cb"
}
```

**Comment by yanxue-intel** [comment]
> For my understanding, is the sandbox key provided in this documentation generic across all Intel pre-production platforms or is it specific to the ICX platform? i.e. does 6.2 work for any processor?

Sandbox key is the same on all Intel pre-production platforms.

File: docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md (hunk 2 above)
Comment (yanxue-intel) [reply docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md:308]: Got it, Thanks

File: docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md (hunk 1 above)
Comment (yanxue-intel) [reply docs/GettingStartedDocs/Contributors/NonAccMachineSGXLinuxGettingStarted.md:311]: OK, I will add it.
